mycpp/gc_alloc.h

```diff
@@ -22,6 +22,40 @@ extern MarkSweepHeap gHeap;
 #define VALIDATE_ROOTS 0
 
 // mycpp generates code that keeps track of the root set
+class StackRoot {
+ public:
+  StackRoot(void* root) {
+#if VALIDATE_ROOTS
```

Comment:
Hm seems like we can extract a common ValidateRoot() free function, since it doesn't use `this` ?

It could take either `void*` or `RawObject*`, I think `RawObject*`  does give some type safety in some cases

Reply:
Ah sure. `RawObject*` seems logical

Reply:
Pushed